extras/gizmo/axis-shapes.js

```diff
@@ -0,0 +1,407 @@
+    }
+
+    get size() {
+        return this._size;
+    }
+}
+
+class AxisBoxLine extends AxisShape {
+    _gap = 0;
+
+    _lineThickness = 0.04;
+
+    _lineLength = 0.5;
+
+    _boxSize = 0.14;
+
+    constructor(options = {}) {
+        super(options);
+
+        this._createBoxLine(options.layers ?? []);
+    }
+
+    set gap(value) {
+        this._gap = value ?? 0;
+        this._updateLine();
+        this._updateBox();
+    }
+
+    get gap() {
+        return this._gap;
+    }
+
+    set lineThickness(value) {
+        this._lineThickness = value ?? 1;
+        this._updateLine();
+        this._updateBox();
+    }
+
+    get lineThickness() {
+        return this._lineThickness;
+    }
+
+    set lineLength(value) {
+        this._lineLength = value ?? 1;
+        this._updateLine();
+        this._updateBox();
+    }
+
+    get lineLength() {
+        return this._lineLength;
+    }
+
+    set boxSize(value) {
+        this._boxSize = value ?? 1;
+        this._updateBox();
+    }
+
+    get boxSize() {
+        return this._boxSize;
+    }
+
+    _createBoxLine(layers) {
+        this.entity = new Entity('axis_' + this.axis);
+        this.entity.setLocalPosition(this._position);
+        this.entity.setLocalEulerAngles(this._rotation);
+        this.entity.setLocalScale(this._scale);
+
+        this._line = new Entity('line_' + this.axis);
+        this._line.addComponent('render', {
+            type: 'cylinder',
+            layers: layers,
+            material: this._defaultColor,
+            castShadows: false
+        });
+        this._updateLine();
+        this.entity.addChild(this._line);
+        this.meshInstances.push(...this._line.render.meshInstances);
+
+        this._box = new Entity('box_' + this.axis);
+        this._box.addComponent('render', {
+            type: 'box',
+            layers: layers,
+            material: this._defaultColor,
+            castShadows: false
+        });
+        this._updateBox();
+        this.entity.addChild(this._box);
+        this.meshInstances.push(...this._box.render.meshInstances);
+    }
+
+    _updateLine() {
+        this._line.setLocalPosition(new Vec3(0, this._gap + this._lineLength * 0.5, 0));
+        this._line.setLocalScale(new Vec3(this._lineThickness, this._lineLength, this._lineThickness));
+    }
+
+    _updateBox() {
+        this._box.setLocalPosition(new Vec3(0, this._gap + this._boxSize * 0.5 + this._lineLength, 0));
+        this._box.setLocalScale(new Vec3(this._boxSize, this._boxSize, this._boxSize));
+    }
+}
+
+class AxisDisk extends AxisShape {
+    _device;
+
+    _tubeRadius = 0.02;
+
+    _ringRadius = 0.55;
+
+    constructor(options = {}) {
+        super(options);
+
+        this._device = options.device;
+        this._ringRadius = options.ringRadius ?? this._ringRadius;
+        this._createDisk(options.layers ?? []);
+    }
+
+    _createDisk(layers) {
+        const mesh = createTorus(this._device, {
```

Comment:
Can we control the number of segments here either directly, or infer based on the size? It's more noticeable when the scale is large.

<img width="330" alt="image" src="https://github.com/playcanvas/engine/assets/430764/c4023782-3f0b-4c7b-8469-5439fa6d6b51">

Reply:
I will check the createTorus method but I should be able to